libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -193,16 +201,24 @@ internal class MapboxTripSession(
     /**
      * Start MapboxTripSession
      */
-    override fun start() {
+    override fun start(withTripService: Boolean) {
```

Comment:
lets toggle `withTripService` to **false** when `TripSession` is stopped and add tests for that case

Reply:
Why is this needed?

Reply:
`fun isRunningForegroundService()` based on that return if a service is running. When a session is stopped -> the service is not running 😃

Reply:
We shouldn't default `withTripService` to `false`. The default behavior should run the foreground service as that is the requirement for most of the mobile use cases.

What we might do instead though, is first check if the trip session is started before returning if the service is running:
```kotlin
    fun isTripSessionRunningService(): Boolean {
        return tripSession.isStarted() && tripSession.isRunningForegroundService()
    }
```

Reply:
looks good to me

Reply:
nit: I'd replace `isRunningForegroundService` -> `isRunningWithForegroundService`

Reply:
Yep I agree we should neither set `withTripService ` to false nor toggle `withTripService` to false when `TripSession` is stopped